packages/taquito/src/contract/errors.ts

```diff
@@ -32,6 +32,18 @@ export class InvalidDelegationSource extends ParameterValidationError {
   }
 }
 
+/**
+ *  @category Error
+ *  @description Error that indicates an invalid staking source contract address being passed or used
+ */
+export class InvalidStakingSource extends ParameterValidationError {
+  constructor(public readonly source: string) {
+    super();
+    this.name = `InvalidStakingSource`;
+    this.message = `The source for staking operations can not be a contract address ${source}.`;
```

Comment:
contract in tezos sense, can also be implicit might be better to use `originated address` or `smart contract address`?

Reply:
I agree with An here, we should explicitly specify which contract it is.

originated contracts: `KT1...`
implicit contracts: `tz1...`, `tz2...`, etc

Reply:
bump